File: recrutplus-backend/src/main/java/com/recrutplus/recrutplus/model/JobOffer.java
package com.recrutplus.recrutplus.model;

import com.recrutplus.recrutplus.model.enums.ContractType;
import com.recrutplus.recrutplus.model.enums.Specialty;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "job_offer")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class JobOffer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Enumerated(EnumType.STRING)
    @Column(name = "specialty", nullable = false)
    private Specialty specialty;

    @Enumerated(EnumType.STRING)
    @Column(name = "contract_type", nullable = false)
    private ContractType contractType;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "address_id", referencedColumnName = "id")
    private Address address;

    @Column(name = "salary")
    private Double salary;

    @Column(nullable = false)
    private Boolean isActive = true;

    @Column(nullable = false, updatable = false)
    private LocalDateTime creationDate;

    private LocalDateTime updatedAt;

    @OneToMany(mappedBy = "jobOffer", cascade = CascadeType.ALL)
    private List<Application> applications;
}

File: recrutplus-frontend/src/types/JobOffer.ts
import type { Address } from './Address';

export enum Specialty {
  IT = 'IT',
  FINANCE = 'FINANCE',
  MARKETING = 'MARKETING',
  RH = 'RH',
  COMMERCIAL = 'COMMERCIAL',
  LOGISTIQUE = 'LOGISTIQUE',
  SANTE = 'SANTE',
  EDUCATION = 'EDUCATION'
}

export enum ContractType {
  CDI = 'CDI',
  CDD = 'CDD',
  STAGE = 'STAGE',
  ALTERNANCE = 'ALTERNANCE'
}

export interface JobOffer {
  id: number;
  title: string;
  specialty: Specialty;
  contractType: ContractType;
  content: string;
  address?: Address;
  salary?: number;
  isActive: boolean;
  creationDate: string;
  updatedAt?: string;
  applicationsCount?: number;
}

export interface CreateJobOfferDTO {
  title: string;
  specialty: Specialty;
  contractType: ContractType;
  content: string;
  address?: Address;
  salary?: number;
}

export interface UpdateJobOfferDTO {
  title?: string;
  specialty?: Specialty;
  contractType?: ContractType;
  content?: string;
  address?: Address;
  salary?: number;
  isActive?: boolean;
}